src/lib/storage/chunk.cpp

```diff
@@ -110,12 +101,11 @@ void Chunk::finalize() {
   _is_mutable = false;
 
   // Only perform the max_begin_cid check if it hasn't already been set.
-  if (has_mvcc_data() && !_mvcc_data->max_begin_cid) {
+  if (has_mvcc_data() && _mvcc_data->max_begin_cid.load() == MvccData::MAX_COMMIT_ID) {
     const auto chunk_size = size();
-    Assert(chunk_size > 0, "finalize() should not be called on an empty chunk");
-    _mvcc_data->max_begin_cid = CommitID{0};
+    Assert(chunk_size > 0, "finalize() should not be called on an empty Chunk.");
     for (auto chunk_offset = ChunkOffset{0}; chunk_offset < chunk_size; ++chunk_offset) {
-      _mvcc_data->max_begin_cid = std::max(*_mvcc_data->max_begin_cid, _mvcc_data->get_begin_cid(chunk_offset));
+      set_atomic_max(_mvcc_data->max_begin_cid, _mvcc_data->get_begin_cid(chunk_offset));
```

Comment:
There can't be races at this moment, right? Wouldn't it be more efficient to get the max value normally and than set the max atomically after the loop?

Reply:
Right. Changed.